src/lib/tabs/tab-group.spec.ts

```diff
@@ -250,8 +250,41 @@ describe('MatTabGroup', () => {
 
       expect(fixture.componentInstance.handleFocus).toHaveBeenCalledTimes(1);
       expect(fixture.componentInstance.handleFocus)
-        .toHaveBeenCalledWith(jasmine.objectContaining({index: 1}));
+      .toHaveBeenCalledWith(jasmine.objectContaining({index: 1}));
     });
+    
+    it('should contain tab object with uniqueName in focusChange event', () => {      
+      spyOn(fixture.componentInstance, 'handleFocus').and.callThrough();
+      fixture.detectChanges();
+      
+      const tabLabels = fixture.debugElement.queryAll(By.css('.mat-tab-label'));
+      
+      tabLabels[1].nativeElement.click();
+      fixture.detectChanges();
+      
+      expect(fixture.componentInstance.handleFocus).toHaveBeenCalledTimes(1);
+      expect(fixture.componentInstance.focusEvent.tab.uniqueName)
+        .toEqual('tb02');
+    });
+
+    it('should contain tab object with uniqueName in selectedTabChange event', fakeAsync(() => {
+      let component = fixture.componentInstance;
```

Comment:
Similarly, this test is doing more than checking the tab name. It can be simplified as well:

```ts
      fixture.detectChanges();
      const tabGroup = fixture.debugElement.query(By.css('mat-tab-group')).componentInstance;

      // Select the third tab
      tabGroup.selectedIndex = 2;
      tick();

      expect(fixture.componentInstance.selectEvent.tab.uniqueName).toEqual('tb03');
```

Reply:
This is done - but I need to keep `checkSelectedIndex(2, fixture)` so that the `selectedTabChange` is fired.